### PR TITLE
named UOp class "NOP" [run_process_replay]

### PR DESCRIPTION
**Pull request by DKormann**
https://github.com/tinygrad/tinygrad/pull/5444#issuecomment-2227150287

### CODE REVIEW

**Comment by chenyuxyz** [comment]
can you remove UOp.VAR as part of this?

**Comment by geohot** [comment]
Also, run process replay. Otherwise looking good!

**Comment by DKormann** [comment]
process replay has run https://github.com/tinygrad/tinygrad/actions/runs/10111497854/job/27963562272?pr=5728

**Comment by DKormann** [comment]
done. reused NOOP not sure if this is cool

**Comment by geohot** [comment]
Cool!

File: tinygrad/codegen/uops.py [hunk 1]
```diff
@@ -116,6 +111,23 @@ def _min_max(self) -> Tuple[Optional[UOp], Optional[UOp]]:
         return self.const(self.src[0].vmin.arg+self.src[1].vmin.arg), self.const(self.src[0].vmax.arg+self.src[1].vmax.arg)
     return None, None
 
+@dataclass(frozen=True)
+class NOp(UOp):
+  varname:Optional[str] = None
+  src:Tuple[NOp, ...] = tuple()
+  def __repr__(self): return pretty_print(self, lambda x: f"NOp({x.op}, {x.dtype}, arg={x.arg}, src=(%s))")
+  def name(self, name:Optional[str]=None): return NOp(self.op, self.dtype, self.src, self.arg, varname=name)
```
Comment (geohot) [inline tinygrad/codegen/uops.py:119]: Next step is probably to remove this function and rename `varname` to `name`

**Pull request merged**